pkg/steps/cluster_claim.go

```diff
@@ -45,81 +47,91 @@ func (s *clusterClaimStep) Validate() error {
-		"owner":        s.clusterClaim.Owner,
+		"product":      string(clusterClaim.Product),
+		"version":      clusterClaim.Version,
+		"architecture": string(clusterClaim.Architecture),
+		"cloud":        string(clusterClaim.Cloud),
+		"owner":        clusterClaim.Owner,
 	}
-	if err := s.hiveClient.List(ctx, clusterPools, listOption); err != nil {
-		return fmt.Errorf("failed to list cluster pools with list option %v: %w", listOption, err)
+	if err := hiveClient.List(ctx, clusterPools, listOption); err != nil {
+		return nil, fmt.Errorf("failed to list cluster pools with list option %v: %w", listOption, err)
 	}
 
 	l := len(clusterPools.Items)
 	if l == 0 {
-		return fmt.Errorf("failed to find a cluster pool providing the cluster: %v", listOption)
+		return nil, fmt.Errorf("failed to find a cluster pool providing the cluster: %v", listOption)
 	} else if l > 1 {
-		return fmt.Errorf("find %d cluster pools providing the cluster (%v): should be only one", len(clusterPools.Items), listOption)
+		return nil, fmt.Errorf("find %d cluster pools providing the cluster (%v): should be only one", len(clusterPools.Items), listOption)
 	}
 
 	clusterPool := clusterPools.Items[0]
-	claimName := s.jobSpec.ProwJobID
+	claimName := jobSpec.ProwJobID
 	claimNamespace := clusterPool.Namespace
 	// https://github.com/kubernetes/test-infra/blob/d14589b797fae426bb70cea4843fa46be50c6739/prow/pod-utils/decorate/podspec.go#L99-L101
-	jobNameForLabel := s.jobSpec.Job
+	jobNameForLabel := jobSpec.Job
 	if len(jobNameForLabel) > validation.LabelValueMaxLength {
-		jobNameForLabel = strings.TrimRight(s.jobSpec.Job[:validation.LabelValueMaxLength], ".-")
+		jobNameForLabel = strings.TrimRight(jobSpec.Job[:validation.LabelValueMaxLength], ".-")
 	}
 	claim := &hivev1.ClusterClaim{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      claimName,
 			Namespace: claimNamespace,
 			Labels: map[string]string{
 				kube.ProwJobAnnotation: jobNameForLabel,
-				kube.ProwBuildIDLabel:  s.jobSpec.BuildID,
+				kube.ProwBuildIDLabel:  jobSpec.BuildID,
 			},
 		},
 		Spec: hivev1.ClusterClaimSpec{
 			ClusterPoolName: clusterPool.Name,
 			Lifetime:        &metav1.Duration{Duration: 4 * time.Hour},
 		},
 	}
-	if err := s.hiveClient.Create(ctx, claim); err != nil {
-		return fmt.Errorf("failed to created cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
+	if err := hiveClient.Create(ctx, claim); err != nil {
+		return nil, fmt.Errorf("failed to created cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
 	}
 	logrus.Info("Waiting for the claimed cluster to be ready.")
 	claim = &hivev1.ClusterClaim{}
-	if err := wait.Poll(15*time.Second, s.clusterClaim.Timeout.Duration, func() (bool, error) {
-		if err := s.hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: claimName, Namespace: claimNamespace}, claim); err != nil {
+	if err := wait.Poll(15*time.Second, clusterClaim.Timeout.Duration, func() (bool, error) {
```

Comment:
This needs to be a watch.

Reply:
https://github.com/openshift/ci-tools/pull/1874